### PR TITLE
Switch theming API migration to ng update and other fixes

### PR DESCRIPTION
**Pull request by crisbeto**
Split up into several commits that include the following changes:
1. Switches the `themingApi` migration from `ng generate` to `ng update` as discussed. This required some adjustments to the code that is called by the CLI, as well as some changes to the unit test setup.
2. Fixes an issue where external stylesheets weren't being picked up as expected. This one is by @devversion.
3. Updates the migration logic not to migrate unprefixed deprecated variables if there is no Material import. This can technically leave some transient dependencies unmigrated, but since their names are very generic, we can't guarantee that the user is referring to our variable. This commit also expands the list of variables that will be migrated.
4. Based on external feedback, it adds a message at the end of the migration.
5. Fixes an issue where the `@use` statement for the new API would be inserted too far down in the file, if the file only has one `@import` statement which is placed further down too. This results in a compilation error, because `@use` statements have to be at the top.

### CODE REVIEW

**Comment by devversion** [comment]
@googlebot I consent.

File: src/material/schematics/ng-update/migrations/theming-api-v12/theming-api-migration.ts [hunk 1]
```diff
@@ -24,7 +28,19 @@ export class ThemingApiMigration extends DevkitMigration<null> {
...
       }
     }
   }
+
+  /** Logs out the number of migrated files at the end of the migration. */
+  static globalPostMigration(_tree: unknown, context: SchematicContext): void {
+    const count = ThemingApiMigration.migratedFileCount;
+
+    if (count > 0) {
+      context.logger.info(`Migrated ${count === 1 ? `${count} files` : `1 file`} to the ` +
```
Comment (devversion) [inline src/material/schematics/ng-update/migrations/theming-api-v12/theming-api-migration.ts:41]: I think the ternary needs to be inversed.

Comment (devversion) [reply src/material/schematics/ng-update/migrations/theming-api-v12/theming-api-migration.ts:41]: Also, out of curiosity, is there a doc/guide we could link too? that might be nice to see when the migration ran.

**Review by devversion — APPROVED** [review]
👍 One minor comment.

File: src/material/schematics/ng-update/migrations/theming-api-v12/theming-api-migration.ts (hunk 1 above)
Comment (crisbeto) [reply src/material/schematics/ng-update/migrations/theming-api-v12/theming-api-migration.ts]: I don't think that we have a doc about the migration in particular. It'll be mentioned in the changelog.

**Review by jelbourn — APPROVED** [review]
lgtm

**Pull request merged**